festival.py

```diff
@@ -23,41 +23,48 @@ def info():
 
     _festival.info()
 
-def textToWavFile(text):
+def textToWavFile(text, filename=None):
     """Returns a filename for a wav file created
     with the text 'text'
-    
+
     Use this method if you are concerned about removing
     the wav file once you have used it.
     Being given the filename, you are responsible for opening it
     as you see fit, and removing it once done
     """
     if not isinstance(text, TEXT_TYPE):
         text = text.decode('utf-8')
-    
-    return _festival._textToWav(text)
 
-def textToWav(text):
+    if filename:
```

Comment:
a more elegant method may be to pass `filename` to `_festival._textToWav` and then to do the checking on the lower level. That way we can avoid repeating this code twice here.


See https://github.com/techiaith/pyfestival/pull/4/commits/1da95795d25b05d73b26a387cfc3a6889f8a9cbd#diff-386b8867957e2358c867c0357857a4a8R53 where you could change the code to:

```
if (!PyArg_ParseTuple(args, "s|z:_textToWav", &text, &saved_filename)) return NULL;

if (saved_filename != NULL) {
    tmpfile = saved_filename;
} else {
   tmpfile = make_tmp_filename;
}
```